src/aiida_quantumespresso/workflows/xps.py

```diff
@@ -366,7 +376,7 @@ def get_treatment_filepath(cls):
     @classmethod
     def get_builder_from_protocol(
         cls, code, structure, pseudos, core_hole_treatments=None, protocol=None,
-        overrides=None, elements_list=None, options=None,
+        overrides=None, elements_list=None, atoms_list=None, options=None,
         structure_preparation_settings=None, **kwargs
     ):
```

Comment:
> > The other point is that both of these should be provided via specific inputs and then mapped to the proper AiiDA types as needed. That way, these optional inputs don't need to be included at all if they're not needed by the user.

> Could explain more about what this means? There is an input parameter called calc_binding_energy in the spec.inputs

You have defined these in the process spec, yes, however in `get_builder_from_protocol()`, there are no inputs fields in the function itself and instead we must provide both the correction energies and the switch to calculate the binding energies *via* `kwargs` instead - which isn't as easy to use or provide documentation for.